schema/string.go

```diff
@@ -17,18 +17,24 @@ const (
 	stringUUIDVersion = 4
 )
 
-func checkStringConstraints(v string, minLength, maxLength int, t string) error {
+func checkStringConstraints(v string, minLength, maxLength int, pattern *regexp.Regexp) error {
 	if minLength != 0 && len(v) < minLength {
-		return fmt.Errorf("constraint check error: %s:%v %v < minimum:%v", t, v, minLength)
+		return fmt.Errorf("constraint check error: %v %v < minimum:%v", v, len(v), minLength)
 	}
 	if maxLength != 0 && len(v) > maxLength {
-		return fmt.Errorf("constraint check error: %s:%v %v > maximum:%v", t, v, maxLength)
+		return fmt.Errorf("constraint check error: %v %v > maximum:%v", v, len(v), maxLength)
+	}
+	if pattern != nil {
+		match := pattern.MatchString(v)
```

Comment:
```go
if !pattern.MatchString(v) {
    ...
}
```

Reply:
I think we still need to check if compiledRegexp is nil, because the compiledRegexp won't be initialized in some cases.